ra/ra_test.go

```diff
@@ -1425,7 +1426,10 @@ func TestCheckCertificatesPerNameLimit(t *testing.T) {
 	}
 	// Verify it has no sub errors as there is only one bad name
 	test.AssertEquals(t, err.Error(), "too many certificates already issued for: example.com: see https://letsencrypt.org/docs/rate-limits/")
-	test.AssertEquals(t, len(err.(*berrors.BoulderError).SubErrors), 0)
+	var bErr *berrors.BoulderError
+	if errors.As(err, &bErr) {
```

Comment:
I feel like these should have an `else`, or have the assertion below moved out of the conditional. The old tests would panic if the error failed the type coercion, after this diff the tests would just continue on their merry way without failing anything.

Reply:
Good catch. For my fix I've duplicated the style used on line `1424` and `1441` though adjusted for `errors.As`. The assertion now happens outside of the conditional, if err is not the expected type we `t.Errorf`